### ISSUE TITLE
Unable to start project if you import the parent folder in IntelliJ

### ISSUE THREAD

**Issue by Artur-** [vaadin/flow#13556]
### Description of the bug

If you download e.g. the quickstart app from https://vaadin.com/docs/latest/flow/guide/quick-start you get a "my-todo.zip". If you extract this to a folder called `my-todo` you will end up with `my-todo/my-todo` containing the actual project. Now if you happen to open the parent folder in IntelliJ instead of the inner folder your project will not start when doing right click -> `Run  Application.main()`:

```
Caused by: java.lang.IllegalStateException: Failed to determine project directory for dev mode. Directory 'C:\Users\foo\Downloads\my-todo' does not look like a Maven or Gradle project. Ensure that you have run the prepare-frontend Maven goal, which generates 'flow-build-info.json', prior to deploying your application
```

### Expected behavior

The application starts even if you import the parent folder.

A potential solution would be to look at the `java.class.path` as a fallback. At least in this case the first entry is `C:\Users\foo\Downloads\my-todo\my-todo\target\classes` from which the real project folder can be deduced

### Minimal reproducible example

```
mkdir foo
cd foo
npx @vaadin/cli init bar
```
open `foo` in IntelliJ and run `Application.main()`

### Versions

- Vaadin / Flow version: 23.0.5
- Java version: 17

**Comment by mshabarov** [comment]
Why is this labeled as an enhancement? Sounds more like a bug.

**Comment by Artur-** [comment]
This was fixed in #16399
